fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2954,6 +2956,121 @@ public void validateMetaData() {
         validator.validate();
     }
 
+    /**
+     * Validate and repair known potential issues with record keys. Currently, this method is capable of identifying
+     * and repairing the following scenarios:
+     * <ul>
+     *     <li>For record stores in which record splitting is disabled but the {@code omitUnsplitRecordSuffix} flag
+     *         is {@code true}, keys found missing the {@link SplitHelper#UNSPLIT_RECORD} suffix will be repaired.</li>
+     *     <li>For record stores in which record splitting is disabled, but the record key suffix is found to be
+     *         a value other than {@link SplitHelper#UNSPLIT_RECORD}, the {@link FDBStoreTimer.Counts#INVALID_SPLIT_SUFFIX}
+     *         counter will be incremented.
+     *     <li>For record stores in which record splitting is disabled, but the record key is longer than expected,
+     *         the {@link FDBStoreTimer.Counts#INVALID_KEY_LENGTH} counter will be incremented.
+     * </ul>
+     *
+     * @param continuation continuation from a previous repair attempt or {@code null} to start from the beginning
+     * @param scanProperties properties to provide scan limits on the repair process
+     *   record keys should be logged
+     * @return a continuation or {@code null} if the repair has been completed
+     */
+    public CompletableFuture<byte[]> repairRecordKeys(@Nullable byte[] continuation, @Nonnull ScanProperties scanProperties) {
```

Comment:
Should this have a dry-run flag?

Reply:
I had started to add one, but I had also tried to have it log everything that it found and it got to be a lot of code to manage not "over-logging", but now that you raise this again, I realized that I could at least still have it incrementing the counters but not doing the correction. I'll add it back.

Reply:
Done.